mono/mini/mini-llvm.c

```diff
@@ -8636,42 +8787,9 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
 		case OP_XEXTRACT_I32:
 		case OP_XEXTRACT_I64:
 		case OP_XEXTRACT_R8:
-		case OP_XEXTRACT_R4: {
-			LLVMBasicBlockRef bbs [64];
-			LLVMValueRef switch_ins;
-			LLVMValueRef phi_values [64];
-			int nelems = LLVMGetVectorSize (LLVMTypeOf (lhs));
-			int i;
-
-			g_assert (nelems <= 64);
-			for (i = 0; i < nelems; ++i)
-				bbs [i] = gen_bb (ctx, "XEXTRACT_CASE_BB");
-			cbb = gen_bb (ctx, "XEXTRACT_COND_BB");
-
-			switch_ins = LLVMBuildSwitch (builder, LLVMBuildAnd (builder, rhs, const_int32 (0xf), ""), bbs [0], 0);
-			for (i = 0; i < nelems; ++i) {
-				LLVMAddCase (switch_ins, LLVMConstInt (LLVMInt32Type (), i, FALSE), bbs [i]);
-				LLVMPositionBuilderAtEnd (builder, bbs [i]);
-				phi_values [i] = LLVMBuildExtractElement (builder, lhs, LLVMConstInt (LLVMInt32Type (), i, FALSE), "");
-				LLVMBuildBr (builder, cbb);
-			}
-
-			LLVMPositionBuilderAtEnd (builder, cbb);
-			values [ins->dreg] = LLVMBuildPhi (builder, LLVMTypeOf (phi_values [0]), "");
-			LLVMAddIncoming (values [ins->dreg], phi_values, bbs, nelems);
-
-			MonoTypeEnum type = (MonoTypeEnum)ins->inst_c0;
-			switch (type) {
-			case MONO_TYPE_U1:
-			case MONO_TYPE_U2:
-				values [ins->dreg] = LLVMBuildZExt (ctx->builder, values [ins->dreg], LLVMInt32Type (), "");
-				break;
-			default:
-				break;
-			}
-			ctx->bblocks [bb->block_num].end_bblock = cbb;
+		case OP_XEXTRACT_R4:
+			values [ins->dreg] = LLVMBuildExtractElement (builder, lhs, rhs, "");
```

Comment:
Is this really not needed ? Maybe older llvm versions couldn't handle it or generated bad code ?

Reply:
@vargaz looks like `extractelement` could always accept a variable as an index: https://godbolt.org/z/qK4aic

a quick test:
```csharp
public static void Main()
{
    Console.WriteLine(Sse41.Extract(Vector128.Create(1, 2, 3, 4), ToVar(1)));
    Console.WriteLine(Sse41.Extract(Vector128.Create(1, 2, 3, 4), ToVar(5)));
    Console.WriteLine(Sse41.Extract(Vector128.Create(1, 2, 3, 4), ToVar(128)));
    Console.WriteLine(Sse41.Extract(Vector128.Create(1, 2, 3, 4), ToVar(129)));
    Console.WriteLine(Sse41.Extract(Vector128.Create(1, 2, 3, 4), ToVar(255)));
}

[MethodImpl(MethodImplOptions.NoInlining)]
static byte ToVar(byte i) => i;
```
Returns exactly the same values as CoreCLR

Reply:
But yeah, the C/C++ mnemonic `_mm_extract_epi32` doesn't accept variables (has constexpr)